src/storage/v2/disk/storage.cpp

```diff
@@ -949,6 +949,20 @@ Result<EdgeAccessor> DiskStorage::DiskAccessor::CreateEdge(VertexAccessor *from,
   return EdgeAccessor(edge, edge_type, from_vertex, to_vertex, storage_, &transaction_);
 }
 
+std::optional<EdgeAccessor> DiskStorage::DiskAccessor::FindEdge(Gid gid, View view, EdgeTypeId edge_type,
+                                                                VertexAccessor *from_vertex,
+                                                                VertexAccessor *to_vertex) {
+  auto res = from_vertex->OutEdges(view, {edge_type}, to_vertex);
```

Comment:
yes here is the implementation that worries me if we replicate with a supernode in the system

Reply:
Since for every delta we can have a linear search -> `O(n^2)` again

Reply:
Our usage during replication handling is recreating the delta creation for REPLICA, the view is NEW so would do very limited processing of the delta chain even for supernodes.

Reply:
All written deltas will have the same commit_timestamp + command_id, hence only one delta would be read at most during FindEdge.
```    
if (view == View::NEW && ts == commit_timestamp && cid <= transaction->command_id) {
    break;
}
```